dds/DCPS/transport/framework/TransportRegistry.cpp

```diff
@@ -716,9 +714,44 @@ TransportRegistry::create_new_transport_instance_for_participant(DDS::DomainId_t
 
   inst->load(ach, sect_key);
   config->instances_.push_back(inst);
+
+  dynamically_created_config_map_[instance_config_name] = config;
+  dynamically_created_instance_map_[instance_config_name] = inst;
+
   return true;
 }
 
+void TransportRegistry::delete_dynamically_created_transport(const OPENDDS_STRING& name)
+{
+  if (dynamically_created_config_map_.count(name) == 0) {
+    ACE_ERROR((LM_ERROR,
+               ACE_TEXT("(%P|%t) ERROR: TransportRegistry::")
+               ACE_TEXT("delete_dynamically_created_transport ")
+               ACE_TEXT("could not find a transport config for%C\n"),
+               name.c_str()));
+    return;
+  }
+
+  if (dynamically_created_instance_map_.count(name) == 0) {
+    ACE_ERROR((LM_ERROR,
+               ACE_TEXT("(%P|%t) ERROR: TransportRegistry::")
+               ACE_TEXT("delete_dynamically_created_transport ")
+               ACE_TEXT("could not find a transport instance for%C\n"),
+               name.c_str()));
+    return;
+  }
+
+  remove_config(dynamically_created_config_map_[name]);
+  remove_inst(dynamically_created_instance_map_[name]);
```

Comment:
Should this method also remove entries from these maps?

Reply:
+1 removing entries (find() up front & erase() at the end)

Reply:
I reworked this so that dynamically_created_config_map_ and dynamically_created_instance_map_ are no longer needed.